.github/workflows/clang-format.yml

```
@@ -0,0 +1,27 @@
+name: Clang Format Check
+
+on:
+  push:
+    branches: [ "release", "develop", "freeze", "feature/**" ]
+  pull_request:
+    branches: [ "release", "develop", "freeze", "feature/**" ]
+
+jobs:
+  clang-format:
+    name: Check Clang Format
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Install clang-format
+        run: sudo apt-get install -y clang-format
+      
+      - name: Run clang-format in dry mode
+        run: |
+          FILES=$(find src/ -type f -regex '.*\.\(c\|cpp\|cc\|h\|hpp\)$')
+          echo "Checking formatting on:"
+          echo "$FILES"
+          echo "$FILES" | xargs clang-format --dry-run --Werror
```

```suggestion
          find src -type f -regex '.*\.\(c\|cpp\|cc\|h\|hpp\)$' -exec clang-format --dry-run -Werror "{}" +
```

This would avoid the pitfalls around filenames with spaces; If you want to still print all the files to be checked you would need to run a second `find` like
```suggestion
          echo "Checking formatting on:"
          find src -type f -regex '.*\.\(c\|cpp\|cc\|h\|hpp\)$'
          find src -type f -regex '.*\.\(c\|cpp\|cc\|h\|hpp\)$' -exec clang-format --dry-run -Werror "{}" +
```